hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1414,12 +1422,12 @@ int BleObject::Observer::stopScanning() {
 
 ble_gap_scan_params_t BleObject::Observer::toPlatformScanParams() const {
     ble_gap_scan_params_t params = {};
-    params.extended = 0;
+    params.extended = scanUseCoded_ ? 1 : 0;
     params.active = scanParams_.active;
     params.interval = scanParams_.interval;
     params.window = scanParams_.window;
     params.timeout = scanParams_.timeout;
-    params.scan_phys = BLE_GAP_PHY_1MBPS;
+    params.scan_phys = scanUseCoded_ ? BLE_GAP_PHY_CODED : BLE_GAP_PHY_1MBPS;
```

Comment:
The SoftDevice allow us to use coded PHY and 1MBPS PHY in parallel, not have to pick either of them.

Reply:
I did not know that. Will test after making change to user additional scanParam rather than API.

Reply:
Yes, it appears to work! Feature incorporated.

Reply:
That's awesome!